### ISSUE TITLE
Sticky dropdowns and a (hacky) fix

### ISSUE THREAD

**Issue by opb** [Dogfalo/materialize#1020]
We're using multiple adjacent dropdowns in our nav, with options set to have the dropdown appear on hover, and appear below the menu bar:

```javascript
$(".dropdown-button").dropdown({
    inDuration: 0,
    constrain_width: false,
    belowOrigin: true
});
```

Having the dropdown options (or the 'activates' element, as it's called in the js) below the nav button (or the 'origin') creates problems with the following JS from the library:
```javascript
    // Hover
    if (options.hover) {
      // Hover handler to show dropdown
      origin.on('mouseover', function(e){ // Mouse over
        placeDropdown();
      });

      // Document click handler
      activates.on('mouseleave', function(e){ // Mouse out
        hideDropdown();
      });
    }
```
![screen shot 2015-03-27 at 19 38 50](https://cloud.githubusercontent.com/assets/778850/6875714/00d7a86a-d4b9-11e4-8a61-3b87fc74bfae.png)

The `hideDropdown()` function is only fired if you leave the dropdown menu itself. When the menu is **on top** of the origin, this isn't a problem. But because my menu is **below** the nav button, I can mouse along my navbar, triggering all the `placeDropdown()` functions, but never firing any `hideDropdown()` functions, because I never went over the menu itself in order to trigger the `mouseleave` event shown above. 

So my hacky fix adds variables to track if the origin and/or the menu itself was hovered over. **Basically, if you leave the origin or the menu itself, and you haven't moved onto the other one of those two, then the dropdown hides.** I also had to add a delay on the `hideDropdown()` function because it seems that you mouseleave one of the elements before you mouseover the other (makes sense I guess).

```javascript
// Hover
    if (options.hover) {
      // Hover handler to show dropdown

      var overOrigin = false;
      var overMenu = false;

      origin.on('mouseover', function(e){ // Mouse over
        overOrigin = true;
        placeDropdown();
      });

      origin.on('mouseleave', function(e){ // Mouse over
          overOrigin = false;
          setTimeout(function(){
              if(!overMenu) hideDropdown();
          }, 50);
      });

      activates.on('mouseover', function(e) { // Mouse over
        overMenu = true;
      });

      // Document click handler
      activates.on('mouseleave', function(e){ // Mouse out
        overMenu = false;
          setTimeout(function(){
              if(!overOrigin) hideDropdown();
          }, 50);
      });
    }
```

So, it's hacky, but it does the job of resolving sticky dropdowns when you are using hover and when the dropdown is below the origin. Not submitted a PR because this is ovbiously sub-optimal.

**Comment by Dogfalo** [comment]
Can you also paste in your html code?

**Comment by kampfkeks** [comment]
Well you could add an extra option in the function

**Comment by opb** [comment]
Thanks for the responses. Nav HTML:

```html
<nav class="primary-color">
                <div class="nav-wrapper">
                    <a href="" class="brand-logo">My Site</a>
                    <ul id="nav-mobile" class="right">
                        <li>
                            <a class="dropdown-button" href="#!" data-activates="nav-docs">
                                Docs<i class="mdi-navigation-arrow-drop-down right"></i>
                            </a>
                        </li>
                        <li>
                            <a class="dropdown-button" href="#!" data-activates="nav-things">
                                Things<i class="mdi-navigation-arrow-drop-down right"></i>
                             </a>
                        </li>
                        <li>
                            <a class="dropdown-button" href="#!" data-activates="nav-things2">
                                More Things<i class="mdi-navigation-arrow-drop-down right"></i>
                            </a>
                        </li>
                        <li>
                            <a class="dropdown-button" href="#!" data-activates="nav-users">
                                Users<i class="mdi-navigation-arrow-drop-down right"></i>
                            </a>
                        </li>
                        <li>
                            <a class="dropdown-button" href="#!" data-activates="nav-settings">
                                Settings <i class="mdi-navigation-arrow-drop-down right"></i>
                            </a>
                        </li>
                        <li>
                            <a href="http://myapp.dev/auth/logout">Logout</a>
                        </li>
                    </ul>
                </div>
            </nav>
```

Dropdowns:

```html
<!-- Dropdown Structure -->
<ul id="nav-things" class="dropdown-content">
    <li>
        <a href="http://myapp.dev/admin/batches">Batches</a>
    </li>
    <li>
        <a href="http://myapp.dev/admin/job-types">Job Types</a>
    </li>
</ul>            
<ul id="nav-users" class="dropdown-content">
    <li>
        <a href="http://myapp.dev/admin/users">Users</a>
    </li>
    <li>
        <a href="http://myapp.dev/admin/user/create">Create User</a>
    </li>
    <li>
        <a href="http://myapp.dev/admin/invitations/new">Invite User</a>
    </li>
    <li>
        <a href="http://myapp.dev/admin/invitations">Invitations</a>
    </li>
</ul>            
<ul id="nav-things2" class="dropdown-content">
    <li>
        <a href="http://myapp.dev/things">Index</a>
    </li>
    <li>
        <a href="http://myapp.dev/place/logs">Logs</a>
    </li>
    <li>
        <a href="http://myapp.dev/place/create">Create Thing</a>
    </li>
    <li>
        <a href="http://myapp.dev/things/manage">Manage Thing</a>
    </li>
</ul>                  
<ul id="nav-settings" class="dropdown-content">
    <li>
        <a href="http://myapp.dev/auth/change-password">My profile</a>
    </li>
</ul>        
<ul id="nav-docs" class="dropdown-content">
    <li>
        <a href="http://myapp.dev/docs/doc">OMG</a>
    </li>
</ul>
<!-- end dropdown structure -->
```

Thanks again

**Comment by Dogfalo** [comment]
Fixed since 5b5ba8627cdb7670924e871e15c1e85dfa2652cb
